### PR TITLE
Make dartpad work at runtime when built with dart 2.0 dart2js

### LINKED ISSUE #831: travis cron for dart-pad has started failing
https://travis-ci.org/dart-lang/dart-pad/jobs/400034357

Problems of this form:
```
$ ./tool/travis.sh
/home/travis/.nvm/versions/node/v8.9.1/bin/vulcanize -> /home/travis/.nvm/versions/node/v8.9.1/lib/node_modules/vulcanize/bin/vulcanize
+ vulcanize@1.16.0
added 40 packages in 3.536s
file:///home/travis/.pub-cache/hosted/pub.dartlang.org/collection-1.14.9/lib/src/canonicalized_map.dart:123:41: Error: The method 'retype' isn't defined for the class 'dart.core::Map<#lib1::CanonicalizedMap::C, #lib2::Pair<#lib1::CanonicalizedMap::K, #lib1::CanonicalizedMap::V>>'.
Try correcting the name to the name of an existing method, or defining a method named 'retype'.
  Map<K2, V2> retype<K2, V2>() => _base.retype<K2, V2>();
                                        ^
```

### PR DESCRIPTION
**Pull request by jcollins-g**
Fixes #831, #832.

This includes changes to generated files because updating package:rpc to work with Dart 2.0 is not a small task.

I included a grind target to pull new versions of route.dart into third_party since that's discontinued.  You can see the branch differences here to the last published version:  https://github.com/dart-archive/route.dart/compare/master...jcollins-g:dart2-route?expand=1

### CODE REVIEW

File: pubspec.yaml [hunk 1]
```diff
@@ -3,18 +3,21 @@ author: Dart Team <misc@dartlang.org>
...
   _discoveryapis_commons: '>=0.1.0 <0.2.0'
-  codemirror: ^0.4.1
+  codemirror: '^0.4.2+5.13.4'
   haikunator: ^0.1.0
   http: '>=0.11.1 <0.12.0'
   logging: '>=0.9.0 <0.12.0'
   markdown: ^2.0.0
-  route_hierarchical: ^0.7.0
+  route_hierarchical:
+    path: ../route.dart/
```
Comment (devoncarew) [inline pubspec.yaml]: `../third_party/route.dart/`?

File: tool/travis.sh [hunk 2]
```diff
@@ -16,4 +16,4 @@ dart tool/grind.dart buildbot
 
 # Run the UI/web tests as well.
 # TODO: Our bot is flakey...
-#dart tool/grind.dart test-web
+dart tool/grind.dart test-web
```
Comment (devoncarew) [inline tool/travis.sh]: just confirming that this is intentional

File: third_party/pkg/route.dart/.gitignore [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+.buildlog
+build/
+/node_modules/
+/test/out/
+/npm-debug.log
+.DS_Store
+dart-sdk
+dartium
+.packages
+.analysis_options
```
Comment (devoncarew) [inline third_party/pkg/route.dart/.gitignore]: I would delete this line

File: pubspec.yaml (hunk 1 above)
Comment (devoncarew) [reply pubspec.yaml]: nevermind - updated in the next commit

**Comment by jcollins-g** [comment]
This is now ready for submission.

**Pull request merged**